### PR TITLE
DROOLS-4588: Coverage increase

### PR DESCRIPTION
**Pull request by jomarko**
Dependent on #2563

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/MVELExpressionEvaluatorTest.java [hunk 1]
```diff
@@ -36,6 +39,11 @@ public void evaluateUnaryExpression() {
         assertTrue(evaluator.evaluateUnaryExpression("# 1", 1, Integer.class));
         assertFalse(evaluator.evaluateUnaryExpression("# 2", 1, Integer.class));
 
+        assertTrue(evaluator.evaluateUnaryExpression("# " + ACTUAL_VALUE_IDENTIFIER + " == 123", 123, Integer.class));
```
Comment (gitgabrio) [inline drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/MVELExpressionEvaluatorTest.java]: @jomarko 
Hi. I think @danielezonca  is creating a specific class to hold the string constants used throughout the code, so probably that `#` should be replaced. Beside that, I think we are also going to have two different "expression" identifier, one for value creation (the one used in all this tests, I think) and one for method invocation. This latter case seems not covered- even if I am not sure on how to test it. @danielezonca ?

**Review by gitgabrio — CHANGES_REQUESTED** [review]
@jomarko 
I read the very last commit and looks almost fine for me - beside a small detail.
What is not really clear to me is that in the PR made by @danielezonca something is already changed

**Comment by jomarko** [comment]
@gitgabrio @dupliaka @yesamer @danielezonca may I ask you for a re-review?

**Review by danielezonca — APPROVED** [review]

**Comment by jomarko** [comment]
@danielezonca may I ask you for help? As you see in last commit I updated the test to use proper type for particular expressions, however problem is with array expressions like **# {"Jim"}** where I expect the type **Object[]** however getting an exception from class loader that it can not be loaded. Any advice?

**Review by yesamer — APPROVED** [review]
@jomarko fine for me, my only suggestion is to apply the change requested by @gitgabrio during his review.

**Comment by jomarko** [comment]
@yesamer yes, good point, @danielezonca could you please point me to the class with those constants?

**Comment by yesamer** [comment]
> @yesamer yes, good point, @danielezonca could you please point me to the class with those constants?

It should be `ConstantHolder` class

**Comment by danielezonca** [comment]
@jomarko 
I had a look to your failing test and the problem is related to load class of arrays. It is broken/not supported so I have created a ticket for that ([DROOLS-4649](https://issues.jboss.org/browse/DROOLS-4649))

**Review by dupliaka — APPROVED** [review]
Fine for me.

**Comment by jomarko** [comment]
jenkins please retest this

**Review by gitgabrio — APPROVED** [review]

**Pull request merged**